compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -989,13 +989,109 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         can_suggest_clone
     }
 
+    fn suggest_cloning_on_spread_operator(
```

Comment:
`spread_operator`? `functional_record_update`

also: please cmment what this function is doing